src/engine/specification.md

```diff
@@ -119,12 +82,42 @@ This structure maps on the [`ExecutionPayload`](https://github.com/ethereum/cons
 1. `Object` - The state of the fork choice:
 - `headBlockHash`: `DATA`, 32 Bytes - block hash of the head of the canonical chain
 - `finalizedBlockHash`: `DATA`, 32 Bytes - block hash of the most recent finalized block
+- `payloadAttributes`: `Object|None` - instance of [`PayloadAttributes`](#PayloadAttributes) or `None`
 
 #### Returns
-None or an error
+
+`Object|Error` - Either instance of response object or an error. Response object:
```

Comment:
Doesn't appear the errors that can be returned are described?

Reply:
Must we describe all errors? Anything that actually *fails* on the client during this call might bubble up an error. At interop, the general idea was to have return values not error under normal operation (thus SYNCING instead of "error: block missing") and for errors to actually represent errors in the operation.

Peter made it seem like if any number of things fail in geth, they'd surface that here.

Reply:
You can have `SUCCESS` or you could be or enter into `SYNCING`. In the event that something fails outside of those two, you then send an error. 

Can we define all such errors?